trunner/harness/plo.py

```diff
@@ -146,14 +146,24 @@ def erase(self, device: str, offset: int, size: int, timeout: Optional[int] = No
             e.cmd = cmd
             raise e
 
-    def jffs2(self, device: str, erase: bool, cleanmarkers: PloJffs2CleanmarkerSpec, timeout: Optional[int] = None):
+    def jffs2(self, device: str, erase: bool, cleanmarkers: PloJffs2CleanmarkerSpec, block_timeout: int):
         """Performs jffs2 command."""
 
+        block_count = cleanmarkers.number_of_blocks
+
         cmd = f"jffs2 -d {device} -c {cleanmarkers}"
         if erase:
             cmd += " -e"
 
-        self.cmd(cmd, timeout)
+        self.send_cmd(cmd)
+
+        for i in range(0, block_count):
+            try:
+                self.dut.expect_exact(f"jffs2: block {i}/{block_count}", timeout=block_timeout)
+            except pexpect.TIMEOUT as e:
+                raise PloError("Wrong jffs2 command output!", cmd=cmd, output=self.dut.before) from e
```

Comment:
I see that you removed this newline from there, but it would be good to add it before "ERROR WHILE FLASHING DEVICE" as we discussed during the meeting, so I'd opt for:

```
# the newline is needed to avoid printing exception in the same line as plo prompt
print(bold("\nERROR WHILE FLASHING THE DEVICE"))
```